src/ln/channelmonitor.rs

```diff
@@ -516,38 +603,82 @@ impl ChannelMonitor {
 			feerate_per_kw,
 			htlc_outputs,
 		});
-		self.key_storage = if let KeyStorage::PrivMode { ref revocation_base_key, ref htlc_base_key, ref delayed_payment_base_key, ref payment_base_key, ref shutdown_pubkey, ref latest_per_commitment_point, .. } = self.key_storage {
-			KeyStorage::PrivMode {
+		self.key_storage = if let Storage::Local { ref revocation_base_key, ref htlc_base_key, ref delayed_payment_base_key, ref payment_base_key, ref shutdown_pubkey, ref latest_per_commitment_point, ref funding_info, ref short_channel_id, .. } = self.key_storage {
+			Storage::Local {
 				revocation_base_key: *revocation_base_key,
 				htlc_base_key: *htlc_base_key,
 				delayed_payment_base_key: *delayed_payment_base_key,
 				payment_base_key: *payment_base_key,
 				shutdown_pubkey: *shutdown_pubkey,
 				prev_latest_per_commitment_point: *latest_per_commitment_point,
 				latest_per_commitment_point: Some(local_keys.per_commitment_point),
+				funding_info: funding_info.clone(),
```

Comment:
nit: if you make the funding_info ref mut, you can do a take() here, instead, avoiding the malloc implied by the Vec copy inside the Script. This is also true in a number of other places.